src/sqlfluff/dialects/dialect_postgres.py

```diff
@@ -1183,9 +1183,7 @@ class UnorderedSelectStatementSegment(ansi.UnorderedSelectStatementSegment):
 
     match_grammar = ansi.UnorderedSelectStatementSegment.match_grammar.copy()
     match_grammar.terminator = match_grammar.terminator.copy(  # type: ignore
-        insert=[
-            Sequence("ON", "CONFLICT"),
-        ],
+        insert=[Sequence("ON", "CONFLICT"), Ref("WithCheckOptionSegment")],
```

Comment:
Is this still needed? Looks like you aren't using this anymore.

Reply:
Yes this is required see other comment